### PR TITLE
Add missing matplotlib dev requirement to compile docs locally

### PR DESCRIPTION
**Pull request by winni2k**
I had to do this extra step to get the docs compiled locally on my machine.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -128,7 +128,15 @@ and the local development version of msprime.
 (sec_development_documentation_building)=
 ### Building
 
-To build the documentation locally, go to the `docs` directory and run `make`.
+To build the documentation locally, follow the 
+{ref}`Quickstart <sec_development_quickstart>`. Then install the python
+requirements for building the documentation:
+
+```bash
+pip install -r requirements/CI-docs/requirements.txt
```
Comment (jeromekelleher) [inline docs/development.md:136]: Hmm, not sure we want to do this. The CI requirements are all pinned, which will cause issues for some people.

We should just add the packages needed to the ``requirements/development.txt`` - which ones were missing?

**Review by jeromekelleher — COMMENTED** [review]
LGTM, but see comment.

File: docs/development.md (hunk 1 above)
Comment (winni2k) [reply docs/development.md:136]: matplotlib was the one I saw. There might be more. I'll debug and find out.

**Comment by winni2k** [comment]
Ok, it looks like adding matplotlib to the dev requirments is sufficient to build all the docs on my machine.

**Review by jeromekelleher — APPROVED** [review]
LGTM, thanks.

**Pull request merged**